toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -244,7 +244,12 @@ func CreatePartitions(diskDevPath string, disk configuration.Disk, rootEncryptio
 
 	// Create new partition table
 	partitionTableType := disk.PartitionTableType
-	_, stderr, err = shell.Execute("parted", diskDevPath, "--script", "mklabel", partitionTableType.String())
+	partedArgument, err := partitionTableType.ConvertToPartedArgument()
+	if err != nil {
+		logger.Log.Warnf("Unable to convert partition table type to parted argument")
```

Comment:
Maybe also print what was the value we were trying to convert. Either here or inside the method. And shouldn't this be an error, since at this point we're not expecting any issues with this value (and if the user provided an invalid one, then that's also an error)?

Reply:
Sure, I will add a debug-level message prior to the call.

Regarding error vs warn, the primary error message will be returned to the caller immediately in the invalid case here. It is the caller's responsibility to determine what to do next. And I do not want to squash the error message with a new one generated here. The warning is just extra context since the user typically runs at info level, but do you think it is unnecessary?

Reply:
Discussed offline and resolved confusion between `fmt.Error()` and `logger.Log.Errorf()`. New update posted.